pkg/vdr/peer/creator.go

```diff
@@ -7,45 +7,64 @@ SPDX-License-Identifier: Apache-2.0
 package peer
 
 import (
+	"crypto/ed25519"
 	"fmt"
 	"time"
 
 	"github.com/btcsuite/btcutil/base58"
 	"github.com/google/uuid"
+	gojose "github.com/square/go-jose/v3"
 
 	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
 	vdrapi "github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr/create"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr/doc"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
 )
 
 const ed25519VerificationKey2018 = "Ed25519VerificationKey2018"
 
 // Build builds new DID Document.
-func (v *VDR) Build(pubKey *vdrapi.PubKey, opts ...vdrapi.DocOpts) (*did.Doc, error) {
-	docOpts := &vdrapi.CreateDIDOpts{}
+func (v *VDR) Build(keyManager kms.KeyManager, opts ...create.Option) (*did.Doc, error) {
+	docOpts := &create.Opts{}
 	// Apply options
 	for _, opt := range opts {
 		opt(docOpts)
 	}
 
-	didDoc, err := build(pubKey, docOpts)
+	didDoc, err := build(keyManager, docOpts)
 	if err != nil {
 		return nil, fmt.Errorf("create peer DID : %w", err)
 	}
 
 	return didDoc, nil
 }
 
-func build(pubKey *vdrapi.PubKey, docOpts *vdrapi.CreateDIDOpts) (*did.Doc, error) {
+func build(keyManager kms.KeyManager, docOpts *create.Opts) (*did.Doc, error) { //nolint: funlen
+	if len(docOpts.PublicKeys) == 0 {
+		id, pubKeyBytes, err := keyManager.CreateAndExportPubKeyBytes(kms.ED25519Type)
```

Comment:
Why do we have code that automatically creates an ED25519 key?

Reply:
This logic was originally there. I didn't want to change it to not break current logic

Reply:
We can check the logic in different issue

Reply:
Ok - we should create an issue to followup in this milestone - I don't understand where this is used -- and clean this up.